ports/stm32/mboot/main.c

```diff
@@ -707,9 +607,10 @@ int do_page_erase(uint32_t addr, uint32_t *next_addr) {
 void do_read(mboot_addr_t addr, size_t len, uint8_t *buf) {
     #if MBOOT_ENABLE_PACKING
     // Read disabled on packed (encrypted) mode.
+    mboot_state_change(MBOOT_STATE_READ_START, addr);
     dfu_context.status = DFU_STATUS_ERROR_FILE;
     dfu_context.error = MBOOT_ERROR_STR_INVALID_READ_IDX;
-    led0_state(LED0_STATE_SLOW_INVERTED_FLASH);
+    mboot_state_change(MBOOT_STATE_READ_END, -MBOOT_ERRNO_FLASH_READ_DISALLOWED);
```

Comment:
Having a negative operator on a not obviously numeric macro seems odd.

Reply:
This pattern is already used a lot in the code, to negate the errno.  It's an enum.

I guess `mboot_state_change()` should have the second argument being signed...